core/node/libp2p/rcmgr.go

```diff
@@ -61,10 +62,15 @@ func ResourceManager(cfg config.SwarmConfig) interface{} {
 			// is documented in docs/config.md.
 			// Any changes here should be reflected there.
 			if cfg.ResourceMgr.Limits != nil {
-				l := *cfg.ResourceMgr.Limits
-				// This effectively overrides the computed default LimitConfig with any vlues from cfg.ResourceMgr.Limits
-				l.Apply(limitConfig)
-				limitConfig = l
+				userSuppliedOverrideLimitConfig := *cfg.ResourceMgr.Limits
+				// This effectively overrides the computed default LimitConfig with any non-zero values from cfg.ResourceMgr.Limits.
+				// Because of how how Apply works, any 0 value for a user supplied override 
+				// will be overriden with a computed default value.
+				// There currently isn't a way for a user to to supply a 0-value override.
```

Comment:
I think we should add this paragraph to https://github.com/ipfs/kubo/blob/master/docs/config.md#swarmresourcemgrlimits, so users know "0 == autoscaled default"

Reply:
Good call @lidel .  I updated the docs.